pkg/sql/opt/exec/execbuilder/tests/local/local_test: TestExecBuild_explain_analyze failed

pkg/sql/opt/exec/execbuilder/tests/local/local_test.TestExecBuild_explain_analyze [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8865697?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/8865697?buildTab=artifacts#/) on master @ [528cd197bf43793672d0660dcabdd89676c0f753](https://github.com/cockroachdb/cockroach/commits/528cd197bf43793672d0660dcabdd89676c0f753):


```
=== RUN   TestExecBuild_explain_analyze
initialized metamorphic constant "merge-joiner-groups-buffer" with value 12
initialized metamorphic constant "direct-scans-enabled" with value true
initialized metamorphic constant "ColIndexJoin-using-streamer-batch-size" with value 2293447
initialized metamorphic constant "max-batch-size" with value 4202
initialized metamorphic constant "max-batch-byte-size" with value 13216319
initialized metamorphic constant "resolve-membership-single-scan-enabled" with value false
initialized metamorphic constant "copy-batch-size" with value 5430
initialized metamorphic constant "async-IE-result-channel-buffer-size" with value 22
initialized metamorphic constant "addsst-rewrite-concurrency" with value 6
initialized metamorphic constant "addsst-rewrite-forced" with value true
initialized metamorphic constant "logictest-use-mvcc-range-tombstones-for-point-deletes" with value true
I230228 00:47:27.919396 1 (gostd) rand.go:199  [T1] 1  random seed: -1376073015967960271
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/ec6a1341baa02e55a69448c66fcc4415/logTestExecBuild_explain_analyze1566110367
    test_log_scope.go:79: use -show-logs to present logs inline
    logic.go:2937: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/6249/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/explain_analyze:127: EXPLAIN (VERBOSE) SELECT * FROM system.privileges WHERE path = 'vtable/crdb_internal/tables';
        expected:
            distribution: local
            vectorized: true
            ·
            • scan
              columns: (username, path, privileges, grant_options, user_id)
              estimated row count: 10 (missing stats)
              table: privileges@privileges_path_user_id_key
              spans: /"vtable/crdb_internal/tables"-/"vtable/crdb_internal/tables"/PrefixEnd
        but found (query options: "") :
            distribution: local
            vectorized: true
            ·
            • scan
              columns: (username, path, privileges, grant_options, user_id)
              estimated row count: 1 (100% of the table; stats collected <hidden> ago)
              table: privileges@privileges_path_user_id_key
              spans: /"vtable/crdb_internal/tables"-/"vtable/crdb_internal/tables"/PrefixEnd
[00:47:30] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/6249/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/explain_analyze with config local: 8 tests, 1 failures
[00:47:31] --- total progress: 8 statements
--- total: 8 tests, 1 failures
    logic.go:4204: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/ec6a1341baa02e55a69448c66fcc4415/logTestExecBuild_explain_analyze1566110367
--- FAIL: TestExecBuild_explain_analyze (3.66s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExecBuild_explain_analyze.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-24877

--- a/pkg/sql/opt/exec/execbuilder/testdata/explain_analyze
+++ b/pkg/sql/opt/exec/execbuilder/testdata/explain_analyze
@@ -124,14 +124,18 @@ regions: <hidden>
 statement ok
 GRANT SELECT ON crdb_internal.tables TO root;
 
+# For some reason, even though we explicitly disable the automatic stats
+# collection for both user and system tables, rarely we still see that the stats
+# on system.privileges were collected, so we filter out a single line from the
+# output to make the test deterministic.
 query T
-EXPLAIN (VERBOSE) SELECT * FROM system.privileges WHERE path = 'vtable/crdb_internal/tables';
+SELECT * FROM [EXPLAIN (VERBOSE) SELECT * FROM system.privileges WHERE path = 'vtable/crdb_internal/tables']
+  WHERE info NOT LIKE '%estimated row count%';
 ----
 distribution: local
 vectorized: true
 ·
 • scan
   columns: (username, path, privileges, grant_options, user_id)
-  estimated row count: 10 (missing stats)
   table: privileges@privileges_path_user_id_key
   spans: /"vtable/crdb_internal/tables"-/"vtable/crdb_internal/tables"/PrefixEnd

